meetings/2023-04-06.md

```diff
@@ -0,0 +1,96 @@
+* eP: In #504, NS is always acting as server-side client. CSS has the convenience for implicit --- it generates token without asking the user to authn/z. If implementations were to be separate, then all problems in 504 described would come to place.
+* SC: {Summary of the shortcoming of Solid-OIDC for this picture}
+* eP: I'd like to understand if the token is used by NR .. what stops it from using it elsewhere?
+* JM: Assuming-ly, the token receiver is same entity that subscribed to it. Is there case where the receiver wouldn't be the same entity?
+* JM: Can still constrain the token with multi-embedded credentials. Sign the token by the server, saying this is intended audience.. e.g., NS could generate a token put in the audience of the NR and send it over. No DPOP required. Then NR can't re-use it at all.
+* eP: Constraining the audience.
+* SC: Are we considering alternatives to move towards?
+* JM: As an implementer, just do cURL for now.. the trade-off is with cURL hs the same security-boundaries. The only ... NR to have something like UUID in the URL. Instead of having one nice URL - however they're receiving it. In the future we might want to also have new authn scheme so that NR can have URL setup instead of cURL.
+* JM: Is cURL described in the current spec?
+* eP: metnions authn issue
+* eP: It is not presented to the user - used at runtime but not exposed. In Some systems, webhook for example, it can be copied. here it is in the subscription response but not to human interaction.
+* JM: As I understand it, the URL is response. That's the complex URL?
+* eP: Yes.
+* JM: The developer for app receiving it, they wouldn't be able to setup an endpoint my/app/parameteruuid and feed into library. Not difficult but something to think about.
+* eP: If you compare to authn/z in place, with tokens, setting some access controls on the endpoint.. this might be easier ot have a lightweight receiver and have all this authn/z system. There is a tradeoff.
+* JM: The solution to authn/z can be implemented in a stateless manner whereas cURLs need state maintained on the NR side. The NR needs to remember the id of the cURL during the subscription phase. Whereas if you use a token the authn is inherit in the token without needing to maintain state. Things from the implementation level. From a spec, cURL has fewer open questions.
+* eP: re state of the URL that needs to maintained, but if you talk about in the previous version of the Webhooks, if you have the sender in the response, you need to keep the sender, and add it to the ACL on the receiver. SubscriptionClient gets the sender, so that NR knows about it, and then when the notification comes, NR checks if it comes from NS.
```

Comment:
```suggestion
* JM: The solution to authn/z can be implemented in a stateless manner, whereas cURLs need state maintained on the NR side. The NR needs to remember the ID of the cURL during the subscription phase, whereas if you use a token the authn is inherent in the token without needing to maintain state. Things from the implementation level. From a spec, cURL has fewer open questions.
* eP: re state of the URL, that needs to be maintained, but if you talk about the previous version of Webhooks, if you have the sender in the response, you need to keep the sender, and add it to the ACL on the receiver. SubscriptionClient gets the sender, so that NR knows about it, and then when the notification comes, NR checks if it comes from NS.
```